service/history/historyEngine.go

```diff
@@ -2246,8 +2242,21 @@ func (e *historyEngineImpl) ResetWorkflowExecution(
 	}
 	if request.GetWorkflowTaskFinishEventId() <= common.FirstEventID ||
 		request.GetWorkflowTaskFinishEventId() >= baseMutableState.GetNextEventID() {
-		return nil, serviceerror.NewInvalidArgument("Workflow task finish ID must be > 1 && <= workflow next event ID.")
+		return nil, serviceerror.NewInvalidArgument("Workflow task finish ID must be > 1 && <= workflow last event ID.")
 	}
+
+	var resetReappyType ResetReapplyType
```

Comment:
Why can't we use the grpc definition?

Reply:
why should we use grpc / protobuf definition all over the place?
existing logic around execution info & execution state & version history already made the internal logic hard to use, e.g. we cannot to OOP, but rather C style programming 
```go
versionHistory.GetLastItem
```
vs
```go
versionhistory.GetLastItem(versionHistory)
```